blockchain/blockchain-wallet-v4-frontend
# Fix(Lockbox connection polling upgrades)

## Description
Addresses a long standing issue in which if a modal was closed while we were polling for a device connection or interacting on the device in anyway, that polling/running saga would never stop and would be left in a broken state. This caused issue with any subsequent device interactions because previous polling was usually still occurring.

## Changes
- Disables Lockbox modals from being closed via `onClickOutside`. All modals need to be explicitly closed by user to ensure we cancel currently running sagas and connection polling properly
- Adds modal headers and close icons to all Lockbox modals
- `initializeNewDeviceSetup`, `updateDeviceFirmware` & `pollForDeviceApp` sagas are now setup as `races` between its own completion and the modal close event.  `finally` blocks are added to each and will clean up its own saga state as needed
- Adds consistent naming to Lockbox modals and its folder structure
- Re-enables Lockbox connection polling & socket messaging console logging

## Change Type
- Bug Fix

## Testing Steps
- Switch between different modals that poll for device connections or have multi-step flows.  Close modals half way between and you should see previous saga/polling cancelled and your new modal connections should function.

## Code Checklist
